src/librbd/ImageCtx.h

```diff
@@ -61,6 +62,43 @@ namespace librbd {
   template <typename> class ResizeRequest;
   }
 
+  class ImageQosStatusSingleton {
```

Comment:
Not really -- a singleton doesn't mean it's shared between processes. What's worse, if you shared a librados cluster connection with multiple different images, they would share a singleton even though they wouldn't necessarily have related limits.